bindings/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -715,33 +731,69 @@ def attach_friend_info_if_present(current_range: Ranges.TreeRange,
                     )
                     ds.WithGlobalFriends(
                         friend_chainsubnames, friend_filenames, friend_alias)
-
-        def build_rdf_from_range(current_range: Ranges.TreeRangePerc) -> TaskObjects:
+                    
+        def build_rdf_from_range(current_range: Ranges.TreeRangeSamplePerc) -> TaskObjects:
             """
             Builds an RDataFrame instance for a distributed mapper.
 
             The function creates a TChain from the information contained in the
             input range object. If the chain cannot be built, returns None.
             """
 
-            clustered_range, entries_in_trees = Ranges.get_clustered_range_from_percs(
+            clustered_range, entries_in_trees = Ranges.get_clustered_range_from_percs_sample(
                 current_range)
 
             if clustered_range is None:
                 return TaskObjects(None, entries_in_trees)
-
+            
             ds = ROOT.RDF.Experimental.RDatasetSpec()
-            # add a sample with no name to represent the whole dataset
-            ds.AddSample(
-                ("", clustered_range.treenames, clustered_range.filenames))
+        
+            filenames_cluster = clustered_range.filenames
+            treenames_cluster = clustered_range.treenames
+            
+            treesInFileMap = {}
+            fileids_cluster = []
+                        
+            for i in range(len(filenames_cluster)):
```

Comment:
This would be more Pythonic
```suggestion
            for filename, treename in zip(filenames_cluster, treenames_cluster):
```